### PR TITLE
CHORE: Remove unneeded if/else condition in post#cook

### PR DESCRIPTION
**Pull request by gdpelican**
Just a couple little cleanups I tripped over in the post model 🌷

### CODE REVIEW

**Comment by SamSaffron** [comment]
This looks fine to me, but going to wait till 2.2 to merge (probably in a week and a half)

File: app/models/post.rb [hunk 1]
```diff
@@ -253,17 +253,11 @@ def cook(raw, opts = {})
 
     options = opts.dup
     options[:cook_method] = cook_method
+    options[:user_id] = self.user.id if self.user
+    # At trust level 3, we don't apply nofollow to links
+    options[:omit_nofollow] = true   if add_nofollow?
```
Comment (tgxworld) [inline app/models/post.rb]: This should be `unless` instead of `if`

Comment (gdpelican) [reply app/models/post.rb]: Oops, fixed 😅

**Comment by gdpelican** [comment]
This probably would be a little bit nicer with something like
```
cooked = post_analyzer.cook raw, opts.dup.merge(
  cook_method:   cook_method,
  user_id:       self.user&.id,
  omit_nofollow: omit_nofollow?
).compact
```

but didn't want to do more changes than necessary, it's just a little tidy-up.

**Comment by SamSaffron** [comment]
I am just worried that this enters the "refactoring" type PR which we explicitly do not allow, it already generated so much work reviewing it (I have looked at this multiple times and so has TGX). I am so sorry but I am closing for now.

**Pull request closed without merging**